### PR TITLE
JSON examples for external OT shelf

### PR DESCRIPTION
**Pull request by italobusi**
JSON examples for external OT shelf based on the figure in PR #121

Pending merge of PR #121 

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

### CODE REVIEW

**Comment by sergiobelotti** [comment]
To make more easy the review of the example, the use case considered is to have three different network instance. 
Considering the picture added in PR #121 in Appendix B, the first network is composed by the external shelf NE only (WDM-TE-Node-1 in JSON code), th second network is composed by the ROADM node only (WDMM-TE-Node-2) and finally the third scenario is the complete one, with both external-shelf and ROADM.

File: json-examples/external-ot-shelf.json [hunk 1]
```diff
@@ -0,0 +1,1195 @@
...
+              },
+              {
+                "otsi-carrier-id": 2
+              }
+            ]
+          }
+        ],
+        "node": [
+          {
+            "node-1": "WDM-TE-Node-2",
```
Comment (sergiobelotti) [inline json-examples/external-ot-shelf.json:464]: "node-1" --> node-id

File: json-examples/external-ot-shelf.json [hunk 2]
```diff
@@ -0,0 +1,1195 @@
...
+              },
+              {
+                "otsi-carrier-id": 2
+              }
+            ]
+          }
+        ],
+        "node": [
+          {
+            "node-1": "WDM-TE-Node-1",
```
Comment (sergiobelotti) [inline json-examples/external-ot-shelf.json:57]: "node-1" --> "node-id"

File: json-examples/external-ot-shelf.json [hunk 3]
```diff
@@ -0,0 +1,1195 @@
...
+                      "otsi-group-ref": "Green OTSiG (Reverse)",
+                      "otsi-ref": 2
+                    }
+                  }
+                ]
+              }
+            ]
+          },
+          {
+            "node-1": "WDM-TE-Node-2",
```
Comment (sergiobelotti) [inline json-examples/external-ot-shelf.json:911]: "node-1" --> "node-id"

File: json-examples/external-ot-shelf.json [hunk 4]
```diff
@@ -0,0 +1,1195 @@
...
+                "ietf-te-topology:te": {
+                  "inter-domain-plug-id": "Awo="
+                }
+              },
+              {
+                "tp-id": "23",
+                "ietf-te-topology:te-tp-id": 23
+              }
+            ],
+            "ietf-te-topology:tunnel-termination-point": [
```
Comment (EstherLerouzic) [inline json-examples/external-ot-shelf.json:87]: I have run an automated json vs yang validation tool and the example is not consistent with the yang:

I think this should be in the object "te", not outside

"ietf-te-topology:te": { 
   "te-node-attributes": {...},
   "tunnel-termination-point": [...]

I also think that "ietf-optical-impairment-topology:ttp-transceiver": should not be inside the "local-link-connectivities" container

Comment (sergiobelotti) [reply json-examples/external-ot-shelf.json:87]: Hi Esther, you're right , the file has to be fixed as you suggested.

File: json-examples/external-ot-shelf.json [hunk 5]
```diff
@@ -0,0 +1,1195 @@
...
+                  "is-allowed": false,
+                  "ietf-optical-impairment-topology:ttp-transceiver": [
+                    {
+                      "transponder-ref": 1,
+                      "transceiver-ref": 1
+                    }
+                  ],
+                  "local-link-connectivity": [
+                    {
+                      "link-tp-ref": 1,
```
Comment (EstherLerouzic) [inline json-examples/external-ot-shelf.json:100]: There are no link with termination point tp-id =1 in this network, it is in the other network, but the the relative path for leafref in ietf-te-topology.yang does not allow to get the idea from another network, it is in the same network.

      leaf link-tp-ref {
        type leafref {
          path "../../../../../nt:termination-point/nt:tp-id";
        }
        description
          "LTP.";
      }

File: json-examples/external-ot-shelf.json [hunk 6]
```diff
@@ -0,0 +1,1195 @@
...
+        "ietf-network-topology:link": [
+          {
+            "link-id": "Add-Drop-Link-1-Forward",
+            "source": {
+              "source-node": "WDM-TE-Node-1",
+              "source-tp": "1"
+            },
+            "ietf-te-topology:te": {
+              "te-link-attributes": {
+                "ietf-optical-impairment-topology;OMS-attributes": [
```
Comment (EstherLerouzic) [inline json-examples/external-ot-shelf.json:207]: ietf-optical-impairment-topology;OMS-attributes -> ietf-optical-impairment-topology:OMS-attributes

File: json-examples/external-ot-shelf.json (hunk 5 above)
Comment (EstherLerouzic) [inline json-examples/external-ot-shelf.json:100]: There is an inconsistency of type: link-tp-ref should be "1", not 1

File: json-examples/external-ot-shelf.json (hunk 1 above)
Comment (italobusi) [reply json-examples/external-ot-shelf.json]: Fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/127/commits/9428184ff351410136b69b4029dfb96debba2ab2

**Review by EstherLerouzic — COMMENTED** [review]
Now this is compliant ! thanks!

**Review by EstherLerouzic — APPROVED** [review]

**Pull request merged**